src/LinAlg/hiopLinSolverSparseCUSOLVER.cpp

```diff
@@ -0,0 +1,936 @@
+// other materials provided with the distribution.
+// iii. Neither the name of the LLNS/LLNL nor the names of its contributors may be used to
+// endorse or promote products derived from this software without specific prior written
+// permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+// OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+// SHALL LAWRENCE LIVERMORE NATIONAL SECURITY, LLC, THE U.S. DEPARTMENT OF ENERGY OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+// AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+// EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Additional BSD Notice
+// 1. This notice is required to be provided under our contract with the U.S. Department
+// of Energy (DOE). This work was produced at Lawrence Livermore National Laboratory under
+// Contract No. DE-AC52-07NA27344 with the DOE.
+// 2. Neither the United States Government nor Lawrence Livermore National Security, LLC
+// nor any of their employees, makes any warranty, express or implied, or assumes any
+// liability or responsibility for the accuracy, completeness, or usefulness of any
+// information, apparatus, product, or process disclosed, or represents that its use would
+// not infringe privately-owned rights.
+// 3. Also, reference herein to any specific commercial products, process, or services by
+// trade name, trademark, manufacturer or otherwise does not necessarily constitute or
+// imply its endorsement, recommendation, or favoring by the United States Government or
+// Lawrence Livermore National Security, LLC. The views and opinions of authors expressed
+// herein do not necessarily state or reflect those of the United States Government or
+// Lawrence Livermore National Security, LLC, and shall not be used for advertising or
+// product endorsement purposes.
+
+/**
+ * @file hiopLinSolverSparseCUSOLVER.cpp
+ *
+ * @author Kasia Swirydowicz <kasia.Swirydowicz@pnnl.gov>, PNNL
+ *
+ */
+
+#include "hiopLinSolverSparseCUSOLVER.hpp"
+
+#include "hiop_blasdefs.hpp"
+
+#include "klu.h"
+#include "cusparse_v2.h"
+#include <string>
+#include <sstream>  
+
+// Error checking utility for CUDA
+template <typename T>
+void check_cuda_error(T result,
+                      char const *const func,
+                      const char *const file,
+                      int const line)
+{
+  if (result) {
+    printf("CUDA error at %s:%d, error# %d\n", file, line, result);
+    exit(EXIT_FAILURE);
+  }
```

Comment:
Consider using HiOp logger instead of printing to stdout in order to maintain xSDK compliance:
```suggestion
  if (result) {
    nlp_->log->printf(hovError, "CUDA error at %s:%d, error# %d\n", file, line, result);
    exit(EXIT_FAILURE);
  }
```

Reply:
By the way, @kswirydo, isn't using `exit` a bit too harsh here?